### DISCUSSION TITLE
Cast types beyond short-text?

### DISCUSSION THREAD

**Discussion by immuneeb** [farcasterxyz/protocol#19] [FIP Stage 1: Ideas]
To confirm, the idea is that in the future there will be different types of Casts beyond short-text? If so, which parties will be able to add a new cast type to the protocol & what is the process for doing so?

**Comment by varunsrin** [comment]
Yes, it is designed to be extensible. 

Adding new types means that every Farcaster hub will now need to replicate and sync this data, which is a non-trivial change.  The benefit of each new addition must be weighed against the cost imposed on all the Hubs in the network. 

New types can be proposed by starting a discussion in this section. We don't have a formal improvement proposal and approval process yet, but will clarify that once v2 launches

**Reply by varunsrin** [reply]
Just opened https://github.com/farcasterxyz/protocol/discussions/71 to address this
